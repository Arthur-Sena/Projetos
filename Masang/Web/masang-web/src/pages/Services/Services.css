:root {
    --background: #f2f2f2;
    --laranja-claro: #f28123;
    --vermelho: #B90404;
    --text-color: #868686;
  }
.Services {
    display: flex;
    flex-direction: column;
    /* justify-content: center; */
    align-items: center;
    background-color: var(--background);
    min-height: 100vh;
    
}
/* CSS header da pagina */
.divTitle {
    margin-top: 2vh;
}

.H1Services {
    color: var(--vermelho);
    /* font-weight: bold; */
    font-family: Helvetica;
}

.linhaGradiente {
    margin-top: -2vh;
    height: 1vh;
    background-image: linear-gradient(to right, var(--vermelho) , var(--laranja-claro));
}

.subTitle {
    margin-top: 2vh;
    width: 80vw;
    align-items: center;
}

.text {
    color: var(--text-color);
}

/* main */
.divMain {
    margin-top: 1.2em;

    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;

    width: 80vw;
    height: 14vh;
    background-image: linear-gradient(to right , var(--laranja-claro), var(--vermelho));
}

.subDiv {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    color: white;
}

/* Cards de Solucoes */
.divCards {
    display: flex;
    flex-direction: row;
    align-items: stretch;
    justify-content: space-between;

    flex-wrap: wrap-reverse;

    margin-top: 2.8em;
    width: 80vw;
}

.card {
    background-color: white;
    border-style: solid;
    border-width: 2px;
    border-color: grey;

    display: flex;
    flex-direction: column;
    align-items: center;

    width: 15.5em;
    /* width: 12em; */
    /* margin-bottom: 2em; */
}

.miniCard {
    color: var(--text-color);

    display: flex;
    flex-direction: column;
    align-items: center;
}

.miniCard h4{
    color: var(--laranja-claro);
    font-weight: 400;
}

.miniCard p{
    font-size: 15px;
    width:  90%;
    text-align: center;

}

.inCard{
    height: 15vh;
    margin-top: 5vh;
}

@media (max-width:1219px){
    .card {
        width: 25em;
        margin-bottom: 2em;

    }
}
@media (max-width:1017px){
    .card {
        width: 20em;
    }
}
@media (max-width:814px){
    .card {
        width: 16em;
    }
}
@media (max-width:650px){
    .card{
        display:flex;
        flex-direction: row;
        justify-content: space-around;
        width: 80vw;
    }

    .card img{
        margin-bottom: 5%;
        margin-left: 5%;
    }

    .divMain{
        height: 16vh;
    }

    .divMain{
        height: 24vh;
    }
}
@media (max-width:360px){
    .divMain {
        width: 85vw;
        height: 26vh;
    }
}