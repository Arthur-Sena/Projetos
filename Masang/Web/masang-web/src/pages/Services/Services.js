import React, { Component } from 'react';
// import { Link} from "react-router-dom";
import { Route, Link, Switch, Redirect, Router} from "react-router-dom";

// import Axios from 'axios';

//Css + images
import './Services.css';
import IconPrincipal from '../../assets/icons/icon-bloodTransfusion.png'; //Icone de doacao de sangue
import IconHospital from '../../assets/icons/hospital.png'; //Icone do banco de sangue
import IconBolsaDeSangue from '../../assets/icons/bloodBag.png'; //Icone de bolsas de sangue
import IconTipoSanguineo from '../../assets/icons/bloodType.png'; //Icone de tipo sanguineo
import IconSangue from '../../assets/icons/bloodSample.png'; //Icone de ... icone
import IconDash from '../../assets/icons/dashBoard.png'; //Icone de dashboard icone


class Services extends Component {

    render() {
        return (
            <div className="Services">

                {/* Titulo da pagina */}
                <div className="divTitle">
                    <h1 className="H1Services">Nossas Soluções</h1>

                    <div  className="linhaGradiente"> </div>
                    {/* <div className="linhaGradiente"  >            </div> */}
                </div>

                {/* Subtitulo da pagina */}
                <div className="subTitle">

                    <p className="text"> Nossa empresa trabalha para que a doação de sangue intensifique-se e cada vez mais vidas possam ser salvas no país. Visando isto, nós fornecemos todas as informações e caminhos que você precisa!</p>

                </div>

                {/*Inicio da Main */}
                <div className="divMain">
                    <div className="subDiv" style={{ width: '40%' }}>
                        <img src={IconPrincipal} style={{ height: '12vh' }} />
                    </div>
                    <div className="subDiv" style={{ width: '45%' }}>
                        <p> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut Lorem ipsum dolor sit </p>
                    </div>
                </div>

                {/* Cards de solucoes */}
                <div className="divCards">

                    <div className="card">
                        <img src={IconHospital} className="inCard" />
                        <div    className="miniCard">
                            {/* <Link to="/Mapa" style={{textDecoration:"none"}}><h4>Buscar Hemocentro</h4></Link> */}
                            <a href="/Mapa" style={{textDecoration:"none"}}><h4>Buscar Hemocentro</h4></a>

                            <p>Veja dados e localizações de todos os hemocentros cadastrados na plataforma</p>
                        </div>
                    </div>

                    <div className="card">
                        <img src={IconTipoSanguineo} className="inCard" />
                        <div    className="miniCard">
                            {/* <Link  to="/CadastroHemocentro" style={{textDecoration:"none"}}><h4>Cadastrar Hemocentro</h4></Link> */}
                            <a href="/CadastroHemocentro" style={{textDecoration:"none"}}> <h4>Cadastrar Hemocentro</h4> </a>

                            <p>Cadastre os dados do hemocentros na plataforma</p>
                        </div>
                    </div>

                    <div className="card">
                        <img src={IconBolsaDeSangue} className="inCard" />
                        <div    className="miniCard">
                        {/* <Link   to="/Campanha"     style={{textDecoration:"none"}}><h4>Listar Campanhas</h4></Link> */}
                        <a href="/Campanha" style={{textDecoration:"none"}}><h4>Listar Campanhas</h4>  </a>

                            <p>Cadastre novas campanhas e liste todas as já cadastradas na plataforma </p>
                        </div>
                    </div>

                    <div className="card">
                        <img src={IconDash} className="inCard" />
                        <div    className="miniCard">                           
                            {/* <Link to="/Dashboard"   style={{textDecoration:"none"}}><h4>Dashboard</h4></Link> */}
                            <a href="/Dashboard"   style={{textDecoration:"none"}}><h4>Dashboard</h4></a>


                            <p>Veja todas as informações da plataforma detalhadamente através de gráficos e tabelas.</p>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}

export default Services