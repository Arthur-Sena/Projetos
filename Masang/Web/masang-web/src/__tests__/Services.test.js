import React from 'react';
import Services from '../pages/Services/Services';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';


//TESTES INSTÂNTANEOS
describe('Services component [SERVICESSCREEN]', () => {

    let component;

    beforeEach(() => {
        component = shallow(<Services />);
    });

    test('deve injetar corretamente um elemento', () => {
        const componenteInput = renderer.create(<input />).toJSON();
        expect(componenteInput).toMatchSnapshot();
    });

    test('deve executar a class e ciclos de vida sem crashar', () => {
        const classServices = mount(<Services />);
        expect(classServices).toMatchSnapshot();
    })

    test('deve executar o componente pai sem crashar', () => {
        expect(component).toMatchSnapshot();
    })

    test('deve renderizar 4 cards com serviços linkados', () => {
        expect(component.find('div[className="card"]').length).toBe(4);
        expect(component.find('a').at(0).prop('href')).toBeDefined();
        expect(component.find('a').at(1).prop('href')).toBeDefined();

        //developing feature by arthur 
        // expect(component.find('a').at(2).prop('href')).toBeDefined();
        // expect(component.find('a').at(3).prop('href')).toBeDefined();
    });

});


//TESTES DE COMPONENTES
// describe.skip('Redirecionamento de Serviços [SERVICESSCREEN]', () => {
//     let componente;

//     beforeEach(() => {
//         componente = shallow(<Services />);
//     });

//     // test('deve encaminhar para busca de hemocentros', async () => {

//     //     const hemolinkText = componente.find('a').at(0);
//     //     const hemocentrolink = componente.find('miniCard').at(0).children().first();

//     //     expect(hemolinkText.html()).toBe('<a href=\"/Mapa\" style=\"text-decoration:none\">a <h4>Buscar Hemocentro</h4></a>');
//     //     hemolinkText.simulate('click');        

//     //     componente.update();
//     //     expect(window.location).toBe('/Mapa');

//     // });


// });