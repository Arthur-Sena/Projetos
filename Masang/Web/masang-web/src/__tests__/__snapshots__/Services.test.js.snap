// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services component [SERVICESSCREEN] deve executar a class e ciclos de vida sem crashar 1`] = `ReactWrapper {}`;

exports[`Services component [SERVICESSCREEN] deve executar o componente pai sem crashar 1`] = `ShallowWrapper {}`;

exports[`Services component [SERVICESSCREEN] deve injetar corretamente um elemento 1`] = `<input />`;
